src/partition/ReceivedPartition.cpp

```diff
@@ -104,7 +104,7 @@ void ReceivedPartition::compute()
       // Filter out vertices not laying in the bounding box
       mesh::Mesh filteredMesh("FilteredMesh", _dimensions, mesh::Mesh::MESH_ID_UNDEFINED);
       // To discuss: maybe check this somewhere in the SolverInterfaceImpl, as we have now a similar check for the parallel case
-      PRECICE_CHECK(!_bb.empty(), "You are running this participant in serial mode and the bounding box on mesh \"{}\", is empty. Did you call setMeshAccessRegion with valid data?", _mesh->getName());
+      PRECICE_CHECK(!_bb.isDefault(), "You are running this participant in serial mode and the bounding box on mesh \"{}\", is empty. Did you call setMeshAccessRegion with valid data?", _mesh->getName());
```

Comment:
I do not understand this change, and it looks like the message is not consistent anymore. Could you please explain your motivation?

Reply:
Maybe it would be helpful to still keep the `empty` function in order to check whether the BB is actually empty or not (which is still different from default).

Reply:
@davidscn Would you be able answer that
1. does the `empty` function here mean "every dimension of the `BoundingBox` is zero"?
2. why should it be changed? The original `empty` function is defined like `isDefault` function.

Reply:
Problem is that the code following this check only works if the bounding box is really non-empty. Example: you call `setMeshAccessRegion`, but only define a single vertex. This will make the box non-default, but the check/code should should stop at this point, because the whole mesh will be filtered afterwards.
How to define empty is a good question. In principle, I guess looking for all dimensions of the BB are zero would be appropriate. I'm not sure if this could lead to issues when running a multi-dimensional coupling and the interface might indeed be 0D. @MakisH any opinions?

Reply:
>  In principle, I guess looking for all dimensions of the BB are zero would be appropriate. I'm not sure if this could lead to issues when running a multi-dimensional coupling and the interface might indeed be 0D. @MakisH any opinions?

I would not worry about issues with mixed dimensions in this PR, and definitely not about 0D. We will revisit the bounding boxes later, if needed.

Keeping both `empty()` and `isDefault()` makes sense to me, I understand that they are different concepts. The `isDefault()` essentially means "I am not initialized - don't use me".

In a 0D simulation, we would probably still have one mesh vertex. The area would be zero, but the box would still have one point.

Reply:
We've (@jeremiahbintang ) added the `empty` method back.
```cpp
bool BoundingBox::empty() const
{
  return (_boundMax - _boundMin).isZero();
}
```
src/partition/ReceivedPartition.cpp
```cpp
PRECICE_CHECK(!_bb.empty(), "You are running this participant in serial mode and the bounding box on mesh \"{}\", is empty. Did you call setMeshAccessRegion with valid data?", _mesh->getName());
```